src/workerd/api/node/zlib-util.c++

```diff
@@ -4,14 +4,105 @@
 // Copyright Joyent and Node contributors. All rights reserved. MIT license.
 
 #include "zlib-util.h"
+#include "workerd/jsg/exception.h"
+
+#include <zlib.h>
+
+#include <iostream>
 
 namespace workerd::api::node {
+uint32_t ZlibUtil::crc32Sync(InputSource data, uint32_t value) {
+  KJ_SWITCH_ONEOF(data) {
+    KJ_CASE_ONEOF(dataBuf, kj::Array<kj::byte>) {
+      return crc32(value, dataBuf.begin(), dataBuf.size());
+    }
+
+    KJ_CASE_ONEOF(dataStr, jsg::NonCoercible<kj::String>) {
+      return crc32(value, dataStr.value.asBytes().begin(), dataStr.value.asBytes().size());
+    }
+  }
 
-uint32_t ZlibUtil::crc32Sync(kj::Array<kj::byte> data, uint32_t value) {
-  // Note: Bytef is defined in zlib.h
-  return crc32(value, reinterpret_cast<const Bytef*>(data.begin()), data.size());
+  KJ_UNREACHABLE;
 }
 
+class GrowableBuffer {
+  // A copy of kj::Vector with some additional methods for use as a growable buffer with a maximum
+  // size
+public:
+  inline explicit GrowableBuffer(size_t _chunkSize, size_t _maxCapacity) {
+    auto maxChunkSize = kj::min(_chunkSize, _maxCapacity);
+    builder = kj::heapArrayBuilder<kj::byte>(maxChunkSize);
+    chunkSize = maxChunkSize;
+    maxCapacity = _maxCapacity;
+  }
+
+  inline size_t size() const {
```

Comment:
nit: explicitly marking these `inline` is probably not very effective here.

Reply:
Yeah I copied and pasted most of this from libkj. I wouldn't personally write it like that

Reply:
Do you want me to get rid of `inline`? I was thinking of just keeping it to match kj::Vector but it's no big deal to remove if it looks wonky to you.

Reply:
Up to you.